benchmark.py

```diff
@@ -265,12 +279,28 @@ def _step():
             for _ in range(self.num_warm_iter):
                 _step()
 
+            if self.cuda_graph:
+                s = torch.cuda.Stream()
+                s.wait_stream(torch.cuda.current_stream())
+                with torch.cuda.stream(s):
+                    for _ in range(3):
+                        _step(sync=False)
+                torch.cuda.current_stream().wait_stream(s)
+
+                g = torch.cuda.CUDAGraph()
+                with torch.cuda.graph(g):
+                    _step(sync=False)
+
             total_step = 0.
             num_samples = 0
-            t_run_start = self.time_fn()
+            t_run_start = self.time_fn(True)
             for i in range(self.num_bench_iter):
-                delta_fwd = _step()
-                total_step += delta_fwd
+                if self.cuda_graph:
+                    g.replay()
```

Comment:
This isn't 100% representative of inference with CUDA Graphs as we're not including copying the new inputs into the input buffer. I'm wondering if we want to confer with @mcarilli to see if we can come up with a reasonable async mechanism to copy the data in. I'm wondering if we want to just do something like have two CUDA Graphs going at the same time so we can async copy one input into one graph while running the other graph and just ping-pong back and forth.

Reply:
This would be important to do if we're actually doing inference and wanting the highest perf possible. For benchmarking we could just keep it simple with running the same inputs over and over.

Reply:
We can add inputs copying like `static_inputs.copy_(self.example_inputs)` before every replay.

Reply:
That'll really slow down the benchmarks as it will be 100% serialized. I'm hoping @mcarilli could suggest a mechanism (though maybe somewhat complex) to be able to overlap it. I'm thinking if we really want having 2 CUDA Graphs around to ping pong between would probably work fine if we throw them in different streams. Not an absolute must, but wanted to make sure we mentioned this added complexity.